### DISCUSSION TITLE
Types & arguments highlighting in rust

### DISCUSSION THREAD

**Discussion by crystalball808** [arcticicestudio/nord#190] [Help & Support] [answered]
Is there a way to make highlighting for types and function arguments for rust?
![image](https://user-images.githubusercontent.com/34882650/227031483-d8b77a58-37e0-4d50-a326-30d85ec0ee8e.png)
Desired way:
![image](https://user-images.githubusercontent.com/34882650/227031751-31c75fd7-e24b-49ab-98d4-06294a0d4a98.png)

**Comment by svengreb** [comment, marked as answer]
Hi @crystalball808 :wave:

These elements are highlighted. [`nord4`, and partly `nord6`][1], are used, among other elements, for variables and parameters. They are highlighted in a decent way to stand out from other elements.
Many other color themes think they need to “kill“ white-ish colors and make all elements look like a _Hello Kitty_ rainbow explosion, e.g. your screenshot from the other theme which make it almost impossible to see on the first glace to which “category“ (keyword, primitive type, custom type etc.) an element belongs to.
Nord has the goal to keep you focus on the code itself while removing any distraction like too many colors that are not consistent across different languages at all.

Anyway, if you‘d still like to change this the [Nord _Visual Studio Code_ port documentation about “customization“][2] includes information and guides how to do so.

[1]: https://www.nordtheme.com/docs/colors-and-palettes#snow-storm
[2]: https://www.nordtheme.com/docs/ports/visual-studio-code/customization

**Reply by crystalball808** [reply]
Thanks for the clarifications. The difference between arguments and variables is not really important to me, but the main reason for opening this issue is color for types, in this case, `Vec` generic. I can provide a reference in Typescript for more clearance.
Here, the `Array` generic is highlighted as a type.
![image](https://user-images.githubusercontent.com/34882650/227192391-5a6afac6-cb0b-44eb-a724-81d0705e0378.png)
Now I can see that I can customize the theme, but maybe this feature would be useful for other developers.

**Reply by svengreb** [reply]
In https://github.com/nordtheme/visual-studio-code/pull/240 the [`entity.name.type` selector][1] has been added which matched a lot of types, but maybe these generic types are handled with another selector.
Feel free to submit a PR to add support for it! The “Customization“ documentation page includes information about the _TextMate_ syntax and references to [Visual Studio Code‘s “Scope Inspector“][2] which is the best way to figure out which element is covered by which selectors.

[1]: https://github.com/nordtheme/visual-studio-code/blob/6138e98f/themes/nord-color-theme.json#L1219
[2]: https://code.visualstudio.com/api/language-extensions/syntax-highlight-guide#scope-inspector
